packages/swc-plugin/src/shared/utils/ast/convertors.rs

```diff
@@ -511,14 +510,13 @@ pub(crate) fn null_to_expression() -> Expr {
 }
 
 pub(crate) fn string_to_prop_name(value: &str) -> Option<PropName> {
-  if IDENT_PROP_REGEX.is_match(value) && value.parse::<i64>().is_err() {
-    Some(PropName::Ident(ident_name_factory(value)))
-  } else {
-    Some(PropName::Str(Str {
+  match Ident::verify_symbol(value) {
+    Ok(_) => Some(PropName::Ident(ident_name_factory(value))),
+    Err(_) => Some(PropName::Str(Str {
       span: DUMMY_SP,
       value: value.into(),
       raw: None,
-    }))
+    })),
```

Comment:
Hi, thanks for reporting the bug and trying to fix it quickly.

Unfortunately, this solution breaks the existing tests, because the string keys of the fields are wrapped in double quotes, which breaks the compatibility of generating CSS classes with the classes of the original library.

```diff
// stylex_transform_stylex_props_test::keep_stylex_create_when_needed::stylex_call_using_exported_styles_with_pseudo_selectors_and_queries
 export const styles = {
<    default: {
>    "default": {
         ":hover_color": "x17z2mba",
         "@media (min-width: 1000px)_backgroundColor": "xc445zv",
         $$css: true
     }
 };
```

I think it is worth trying to improve the current regular expression to something like this:
```rust
pub(crate) static IDENT_PROP_REGEX: Lazy<Regex> =
  Lazy::new(|| Regex::new(r"^[a-zA-Z$_][a-zA-Z\d$_]*$").unwrap());
```

Could you try this solution?

Reply:
You can't reliably check this using regular expressions.

Reply:
Yes, you are absolutely right, it is not the most elegant solution. I suggest to fix the bug first using a regular expression, and in addition you can open an issue or pull request to fix the logic of validation the JSON key. Unfortunately, in the coming days I will not be able to fix it myself, so I will be glad to help🙂